### ISSUE TITLE
NPE when Git change log takes too much time to complete

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#772]
```
java.lang.NullPointerException
	at net.nemerosa.ontrack.common.FutureUtils.wait(FutureUtils.java:22)
	at net.nemerosa.ontrack.common.FutureUtils.wait(FutureUtils.java:17)
	at net.nemerosa.ontrack.extension.git.service.GitServiceImpl.syncAndWait(GitServiceImpl.kt:195)
	at net.nemerosa.ontrack.extension.git.service.GitServiceImpl.changeLog(GitServiceImpl.kt:162)
	at net.nemerosa.ontrack.extension.git.service.GitServiceImpl$$FastClassBySpringCGLIB$$883c1acc.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:750)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:295)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at net.nemerosa.ontrack.extension.git.service.GitServiceImpl$$EnhancerBySpringCGLIB$$7c8c0042.changeLog(<generated>)
	at net.nemerosa.ontrack.extension.git.GitController.changeLog(GitController.kt:162)
```

**Comment by dcoraboeuf** [comment]
If some GitHub indexation jobs in Ontrack are stuck and new jobs cannot be fired any longer, causing `net.nemerosa.ontrack.extension.git.service.GitServiceImpl#sync` to return null, which is not checked (Java vs Kotlin…) in `net.nemerosa.ontrack.common.FutureUtils#wait(java.lang.String, java.util.concurrent.Future<T>)`.

**Comment by dcoraboeuf** [comment]
* Available in 3.43.25
* Available in 4.1.14
